README.md

```diff
@@ -184,16 +184,21 @@ MACOSX_DEPLOYMENT_TARGET=10.9 CC=clang CXX=clang++ python setup.py install
 
 On Windows
 ```cmd
-set "VS150COMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Auxiliary\Build"
+REM The folowing line is for Visual Studio 2017 Enterprise only
+set VS150COMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Auxiliary\Build
+REM The folowing line is for Visual Studio 2017 Professional only
+set VS150COMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Auxiliary\Build
+REM The folowing line is for Visual Studio 2017 Community only
+set VS150COMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build
 set CMAKE_GENERATOR=Visual Studio 15 2017 Win64
 set DISTUTILS_USE_SDK=1
 REM The following two lines are needed for Python 2.7, but the support for it is very experimental.
 set MSSdk=1
 set FORCE_PY27_BUILD=1
 REM As for CUDA 8, VS2015 Update 3 is also required to build PyTorch. Use the following line.
-set "CUDAHOSTCXX=%VS140COMNTOOLS%\..\..\VC\bin\amd64\cl.exe"
+set CUDAHOSTCXX=%VS140COMNTOOLS%..\..\VC\bin\amd64\cl.exe
 
-call "%VS150COMNTOOLS%\vcvarsall.bat" x64 -vcvars_ver=14.11
+call "%VS150COMNTOOLS%\vcvarsall.bat" x64
```

Comment:
Well, if you remove the 14.11 constraint here, then you should add a comment here. Different versions of CUDA rely on different versions of VS. And I can only confirm that (VS 15.3/ VC++14.11)  work for major CUDA versions that we use. (CUDA 9 - CUDA 10)

Reply:
> if you remove the 14.11 constraint here

It is not actually a constraint. With that option user will get the following error:
`[ERROR:vcvars.bat] Toolset directory for version '14.11' was not found.`
for all versions of VS 2017, except Update 3 and 4 (15.3.3, 15.4.4 and 15.4.5). Here is a more or less complete list of all Visual Studio versions (unofficial): [Visual Studio versions](https://dev.to/yumetodo/list-of-mscver-and-mscfullver-8nd). I believe that he/she in case of the above error will try the same as I did: `-vcvars_ver=14.1x`, where `x` is his/her minor toolset version. Finally, I removed this option and built successfully by VS 2017 (15.9.5, toolset 14.16).

> And I can only confirm that (VS 15.3/ VC++14.11) work for major CUDA versions that we use. (CUDA 9 - CUDA 10)

It should be tested with the rest of VS 2017 (15.x) versions and CUDA 9.x-10.0. If the constraint actually takes place - it is bad news, because Visual Studio is being updated actively and there is no mechanism for uninstalling its updates. After that such constraints should be mentioned above in Readme somewhere near "If you want to build on Windows..." or section 'Constraints' should be added.

Reply:
Sounds reasonable. Would you please add it? Some users have tested the conditions for CUDA 9 and CUDA 9.2. https://github.com/pytorch/pytorch/issues/15630#issuecomment-451247858

Reply:
Sorry, what you mean by 'it'? What you think should be added in Readme?

Reply:
I didn't test those configurations so I couldn't guarantee them. Moreover, I have a multiple CUDA  (7.0/7.5/8.0/9.0/9.1/9.2/10.0), and multiple VS (14.0 Update 3 Pro/15.9.5 Pro) installed, and still do not know which CUDA was chosen by pytorch... I'm a new "forced" user of your framework. And what I may guarantee is that the diff in this PR works for me entirely.

The only thing that I may say on the subject is that better to stay in Readme in terms of versions (like 15.x.x), which the user may check easily, than in terms of macros' values. But in your build scripts, you may (but I'd prefer 'should') add your macro based constraints with corresponding warnings and/or errors.

Reply:
> I'm a new "forced" user of your framework. And what I may guarantee is that the diff in this PR works for me entirely.

Yes, but as a developer here, I have to consider for most users. 

> I didn't test those configurations so I couldn't guarantee them.

I pick these combinations out from the code `%CUDA_PATH%\include\crt\host_defines.h`, so they are absolutely true.

> Moreover, I have a multiple CUDA (7.0/7.5/8.0/9.0/9.1/9.2/10.0), and multiple VS (14.0 Update 3 Pro/15.9.5 Pro) installed, and still do not know which CUDA was chosen by pytorch (looks from the build log like the latest 10)

It will depend on your env vars like `PATH` or `CUDA_PATH`, Usually, it will pick the last one your install.

Reply:
CUDA_PATH should be mentioned in Readme as well.